File: julia/word-count/word-count.jl
function cleansentence(sentence)
    commaless = replace(sentence, "," => " ")
    quoteless = replace(commaless, r"\'\s|\s\'" => " ")
    replace(quoteless, r"[^a-z0-9\s\']" => "")
end

function wordcount(sentence)
    counts = Dict()
    words = sentence |> lowercase |> cleansentence |> split
    for w in words
        if haskey(counts, w)
            counts[w] += 1
        else
            w = replace(w, r"^\'|\'$" => "")
            if w == "" || count(i->(i=='''), w) > 1
                continue
            end
            counts[w] = 1
        end
    end
    counts
end

File: python/word-count/word_count.py
import re

def count_words(sentence):
    sentence = sentence.lower().replace(",", " ")
    sentence = re.sub("\'\s|\s\'", " ", sentence)
    words = re.sub("[^a-z0-9\s\']", " ", sentence).split()
    words = [w.strip("'") for w in words]

    counts = {}
    for w in words:
        if w in counts:
            counts[w] += 1
        elif w != "":
            counts[w] = 1
    return counts
